src/Tools/Shared/CommandLine/CommandLineApplicationExtensions.cs

```diff
@@ -21,6 +22,13 @@ public static void OnExecute(this CommandLineApplication app, Action action)
                     return 0;
                 });
 
+        public static void OnExecute(this CommandLineApplication app, Func<Task> func)
```

Comment:
Is there a reason https://github.com/aspnet/AspNetCore/blob/master/src/Mvc/GetDocumentInsider/src/CommandLineUtils/CommandLineApplication.cs#L103 isn't being used? You generally want the app to return the exit code.